homeassistant/components/samsungtv/media_player.py

```diff
@@ -141,57 +87,40 @@ def __init__(self, host, port, name, timeout, mac, broadcast, uuid):
         # Generate a configuration for the Samsung library
         self._config = {
             "name": "HomeAssistant",
-            "description": name,
+            "description": self._name,
             "id": "ha.component.samsung",
-            "method": None,
-            "port": port,
-            "host": host,
-            "timeout": timeout,
+            "method": config_entry.data[CONF_METHOD],
+            "port": config_entry.data.get(CONF_PORT),
+            "host": config_entry.data[CONF_HOST],
+            "timeout": 1,
         }
 
-        # Select method by port number, mainly for fallback
-        if self._config["port"] in (8001, 8002):
-            self._config["method"] = "websocket"
-        elif self._config["port"] == 55000:
-            self._config["method"] = "legacy"
+        # set up config_entry updates listener
+        config_entry.add_update_listener(self.async_config_entry_update)
+
+    @staticmethod
+    async def async_config_entry_update(hass, config_entry):
+        """Pass config_entry updates to component."""
+        registry = await async_get_registry(hass)
+        entity_id = registry.async_get_entity_id(
+            MEDIA_PLAYER_DOMAIN, DOMAIN, config_entry.data.get("id")
+        )
+        component = hass.data[MEDIA_PLAYER_DOMAIN].get_entity(entity_id)
+        if component is not None:
+            await component.update_device_info(config_entry)
+
+    async def update_device_info(self, config_entry):
+        """Update device info according to config_entry."""
+        self._uuid = config_entry.data.get(CONF_ID)
+        self._manufacturer = config_entry.data.get(CONF_MANUFACTURER)
+        self._model = config_entry.data.get(CONF_MODEL)
```

Comment:
This is not the best way, right? And this is also not working at first start. `async_config_entry_update` is getting called when the configured value is not fully loaded, so `component` is `None`.